src/main/java/org/folio/rest/impl/OrdersImpl.java

```diff
@@ -117,9 +119,33 @@ public void putOrdersById(String id, String lang, CompositePurchaseOrder compPO,
 
   @Override
   @Validate
-  public void postOrdersLinesById(String id, String lang, CompositePoLine entity, Map<String, String> okapiHeaders,
+  public void postOrdersLinesById(String id, String lang, PoLine poLine, Map<String, String> okapiHeaders,
                                   Handler<AsyncResult<javax.ws.rs.core.Response>> asyncResultHandler, Context vertxContext) {
-    asyncResultHandler.handle(Future.failedFuture(new NotImplementedException("POST PO line is not implemented yet")));
+    final HttpClientInterface httpClient = getHttpClient(okapiHeaders);
+    PostOrdersHelper helper = new PostOrdersHelper(httpClient, okapiHeaders, asyncResultHandler, vertxContext);
+
+    logger.info("Creating POLine to an existing order...");
+
+    if (poLine.getId() == null) {
+      logger.info("POLine without id. Set id from path url: " + id);
+      poLine.setId(id);
+    }
+
+    if (id.equals(poLine.getId())) {
+      helper.createPoLine(poLine)
+        .thenAccept(pol -> {
+          logger.info("Successfully added PO Line: " + JsonObject.mapFrom(pol).encodePrettily());
+          httpClient.closeClient();
+          Response response
+            = PostOrdersLinesByIdResponse.respond201WithApplicationJson(poLine, PostOrdersLinesByIdResponse.headersFor201());
+          AsyncResult<javax.ws.rs.core.Response> result = Future.succeededFuture(response);
+          asyncResultHandler.handle(result);
+        })
+        .exceptionally(helper::handleError);
```

Comment:
The `PostOrdersHelper#handleError(Throwable)` might return response with `401` status code. However this one is not declared for `POST /orders/{id}/lines` endpoint. This might be confusing.

Reply:
Response 401 dded to RAML.